app/main.py

```diff
@@ -1,32 +1,31 @@
 from __future__ import annotations
+from typing import Any
+from functools import total_ordering
 
 
+@total_ordering
 class Distance:
 
     def __init__(self, km: int | float) -> None:
         self.km = km
 
+    @staticmethod
+    def check_distance(other: Any) -> int | float:
+        if isinstance(other, Distance):
+            return other.km
+        return other
```

Comment:
I guess you should rename function on something like fetch_distance_metric, since we are not checking distance and it's not intuitive )